lib/transaction.js

```diff
@@ -125,7 +129,7 @@ class Transaction {
         throw setupErr;
       }))
       .tap(() => {
-        if (this.sequelize.constructor._cls) {
+        if (useCLS && this.sequelize.constructor._cls) {
```

Comment:
`this.sequelize.constructor._cls` should not set if you dont call `useCls`, can you verify if your test fails without this fix ?

Reply:
Ok understood, I thought we were saving env without cls